crates/sui-types/src/crypto.rs

```diff
@@ -936,10 +954,14 @@ pub trait SuiSignature: Sized + signature::Signature {
 
 impl<S: SuiSignatureInner + Sized> SuiSignature for S {
     fn signature_bytes(&self) -> &[u8] {
+        // Access array slice is safe because the array bytes is initialized as
```

Comment:
`SuiSignature` is a trait, how do we enforce that `as_ref` is always 
`flag || signature || pubkey` with its defined length?